controllers/loadtest_controller_test.go

```diff
@@ -759,3 +762,87 @@ var _ = Describe("Pod Creation", func() {
 		})
 	})
 })
+
+var _ = Describe("getRequeueTime", func() {
+	var test *grpcv1.LoadTest
+	var reconciler *LoadTestReconciler
+
+	BeforeEach(func() {
+		test = newLoadTest()
+		reconciler = &LoadTestReconciler{
+			Log: ctrl.Log.WithName("controllers").WithName("LoadTest"),
+		}
+	})
+
+	Context("when find the start time was newly filled", func() {
+		It("returns the test's timeout", func() {
+			var startTime metav1.Time
+			expectedRequeueTime := time.Duration(test.Spec.TimeoutSeconds) * time.Second
+			previousStatus := grpcv1.LoadTestStatus{}
+
+			startTime.Time, _ = time.Parse("Mon Jan 02 2006 15:04:05 GMT-0700", "Fri Oct 23 2020 15:38:22 GMT-0700")
+			test.Status = grpcv1.LoadTestStatus{StartTime: &startTime}
+
+			requeueTime := getRequeueTime(test, previousStatus, reconciler.Log)
+
+			Expect(requeueTime).To(Equal(expectedRequeueTime))
+		})
+	})
+
+	Context("when find the stop time was newly filled", func() {
+		It("returns the ttl less the actual running time", func() {
+			var startTime metav1.Time
+			var stopTime metav1.Time
+
+			startTime.Time, _ = time.Parse("Mon Jan 02 2006 15:04:05 GMT-0700", "Fri Oct 23 2020 15:38:22 GMT-0700")
+			stopTime.Time, _ = time.Parse("Mon Jan 02 2006 15:04:05 GMT-0700", "Fri Oct 23 2020 15:38:52 GMT-0700")
+
+			previousStatus := grpcv1.LoadTestStatus{StartTime: &startTime}
+			test.Status = grpcv1.LoadTestStatus{StartTime: &startTime, StopTime: &stopTime}
+			expectedRequeueTime := time.Duration(test.Spec.TTLSeconds)*time.Second - test.Status.StopTime.Time.Sub(test.Status.StartTime.Time)
+
+			requeueTime := getRequeueTime(test, previousStatus, reconciler.Log)
+
+			Expect(requeueTime).To(Equal(expectedRequeueTime))
+		})
+	})
+	Context("when neither of start and end time is newly filled", func() {
+		It("returns 0 duration", func() {
+			var startTime metav1.Time
+
+			startTime.Time, _ = time.Parse("Mon Jan 02 2006 15:04:05 GMT-0700", "Fri Oct 23 2020 15:38:22 GMT-0700")
+
+			expectedRequeueTime := time.Duration(0)
+			previousStatus := grpcv1.LoadTestStatus{StartTime: &startTime}
+			test.Status = grpcv1.LoadTestStatus{StartTime: &startTime}
+
+			requeueTime := getRequeueTime(test, previousStatus, reconciler.Log)
+
+			Expect(requeueTime).To(Equal(expectedRequeueTime))
+		})
+
+		It("returns 0 duration", func() {
```

Comment:
It blocks should have different names. This allows us to see what failed quickly. So perhaps these should be more specific, like "returns 0 duration when ...".

Reply:
Right, I thought the conditions has to be in the context...I will modify it.